File: airesources/Java/Move.java
package halitejavabot;

public class Move {
    public enum MoveType {Noop, Thrust, Dock, Undock}
    protected MoveType type;
    protected Ship ship;

    public Move() {
        this.type = MoveType.Noop;
        this.ship = null;
    }

    public MoveType getType() {
        return type;
    }

    public Ship getShip() {
        return ship;
    }

}

File: airesources/CSharp/Halite2/hlt/Move.cs
namespace Halite2.hlt
{
    public class Move
    {
        public enum MoveType { Noop, Thrust, Dock, Undock }

        private MoveType type;
        private Ship ship;

        public Move(MoveType type, Ship ship)
        {
            this.type = type;
            this.ship = ship;
        }

        public MoveType GetMoveType()
        {
            return type;
        }

        public Ship GetShip()
        {
            return ship;
        }
    }
}
